File: backend/src/main/java/com/tripfriend/domain/trip/schedule/dto/TripScheduleUpdateReqDto.java
package com.tripfriend.domain.trip.schedule.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TripScheduleUpdateReqDto {
    @NotBlank
    private String title;

    @NotBlank
    private String description;

    @NotNull
    private LocalDate startDate;

    @NotNull
    private LocalDate endDate;
}

File: backend/src/main/java/com/tripfriend/domain/trip/schedule/dto/TripScheduleUpdateReqDto.kt
package com.tripfriend.domain.trip.schedule.dto

import java.time.LocalDate

class TripScheduleUpdateReqDto @JvmOverloads constructor(
    // 테스트 코드 코틀린 전환 후 변경
    var title: String? = null,

    var description: String? = null,

    var startDate: LocalDate? = null,

    var endDate: LocalDate? = null,
) {}